pkg/lib/common/common.go

```diff
@@ -0,0 +1,23 @@
+package common
+
+import (
+	"time"
+)
+
+const (
+	restartAnnotationName = "istio-operator.kyma-project.io/restartedAt"
+)
+
+func AddRestartAnnotation(annotations map[string]string) map[string]string {
+	if len(annotations) == 0 {
+		annotations = map[string]string{}
+	}
+
+	annotations[restartAnnotationName] = time.Now().Format(time.RFC3339)
+	return annotations
+}
+
+func WasRestarted(annotations map[string]string) bool {
```

Comment:
Maybe we should call this `HasRestart`, since the annotations itself can't be restarted and it aligns better with the `AddRestartAnnotation`.